vlm/src/main/scala/RasterSourceRDD.scala

```diff
@@ -0,0 +1,105 @@
+package geotrellis.contrib.vlm
+
+
+import geotrellis.raster._
+import geotrellis.vector._
+import geotrellis.spark._
+import geotrellis.spark.partition._
+import geotrellis.spark.tiling._
+import geotrellis.proj4._
+
+import org.apache.spark._
+import org.apache.spark.rdd._
+
+
+object RasterSourceRDD {
+  final val PARTITION_BYTES: Long = 128l * 1024 * 1024
+
+  def apply(
+    sources: Seq[RasterSource],
+    layout: LayoutDefinition,
+    partitionBytes: Long = PARTITION_BYTES
+  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] = {
+    val mapTransform = layout.mapTransform
+    val extent = mapTransform.extent
+    val combinedExtents = sources.map { _.extent }.reduce { _ combine _ }
+
+    val cellType = {
+      val cellTypes = sources.map { source => Set(source.cellType) }.reduce { _ ++ _ }
+
+      if (cellTypes.size > 1)
+        throw new Exception(s"All RasterSources must have the same CellType, but multiple ones were found: $cellTypes")
```

Comment:
Should use `require` here, its more idiomatic and will give a more clear exception.